public/app/plugins/panel/flamegraph/components/FlameGraph/FlameGraph.tsx

```diff
@@ -127,6 +133,13 @@ const FlameGraph = ({
     [levels, wrapperWidth, valueField, totalTicks, rangeMin, rangeMax, search, topLevelIndex]
   );
 
+  useEffect(() => {
+    if (levels[topLevelIndex] && levels[topLevelIndex][selectedBarIndex]) {
+      const bar = levels[topLevelIndex][selectedBarIndex];
+      setMetadata(getMetadata(valueField, bar.value, totalTicks));
```

Comment:
This seems to me like a bit weird structuring. The `getMetadata` does not seem to be doing anything super expensive and seems like it needs the things that would make this component rerender anyway so this seems like premature optimization and probably does not give you much. It would be easier to just use `getMetadata` directly in the render function and compute on every render IMHO.

At the same time though the `getMetadata` is already defined in the `<FlameGraphMetadata>` component. Why aren't all the dependencies here just props for the `FlameGraphMetadata` component and the computation done just there? Less things get exported and it would be easy to just add `React.memo` and memorize the component if that needed some perf optimization.

Reply:
Thanks.. I've pushed the changes.